windowsforms/FlexGridShowcaseDemo/cs/MainForm.cs

```diff
@@ -0,0 +1,682 @@
+            greenStyle.ImageAlign = ImageAlignEnum.LeftCenter;
+            greenStyle.ForeColor = Color.Green;
+
+            // add rating style
+            var ratingStyle = _flexGrid.Styles.Add("Rating");
+            ratingStyle.ImageAlign = ImageAlignEnum.RightCenter;
+        }
+
+        private void InitRules()
+        {
+            _rulesManager.SetC1RulesManager(_flexGrid, _rulesManager);
+            var rulesDict = new Dictionary<string, string>()
+                {
+                    { "Discount < 10%", "= [Discount] < 0.1" },
+                    { "Discount < 20%", "= [Discount] < 0.2" },
+                    { "Discount < 30%", "= [Discount] < 0.3" }
+                };
+
+            // creating rules
+            _rules = rulesDict.Keys
+                .Select(x => new C1.Win.RulesManager.Rule()
+                {
+                    Name = x,
+                    Expression = rulesDict[x],
+                    Style = new ItemStyle()
+                    {
+                        ForeColor = Color.FromArgb(_rnd.Next(255), _rnd.Next(255), _rnd.Next(255)),
+                        BorderColor = Color.DarkBlue,
+                        FontStyle = FontStyle.Bold
+                    }
+                });
+
+            // add menu items
+            var ruleRibbonToggleButtons = rulesDict.Keys
+                .Select(x => new RibbonToggleButton()
+                {
+                    Text = x,
+                    Pressed = false
+                });
+
+            foreach (var ruleRibbonToggleButton in ruleRibbonToggleButtons)
+            {
+                ruleRibbonToggleButton.PressedButtonChanged += RuleRibbonToggleButton_PressedButtonChanged;
+                _ribbonMenuFormatting.Items.Add(ruleRibbonToggleButton);
+            }
+        }
+
+        #endregion
+
+        #region Themes
+
+        private void InitThemes()
+        {
+            // register custom theme
+            var customThemePath = Path.Combine(Directory.GetCurrentDirectory(), _customThemeName + ".c1themez");
+            if (File.Exists(customThemePath))
+            {
+                C1ThemeController.RegisterTheme(customThemePath);
+            }
+
+            // load themes into ribbon combo box
+            var themes = C1ThemeController.GetThemes().Where(x => x == _customThemeName || x.Contains("Office2016") || x.Contains("Material"));
+            foreach (var theme in themes)
+            {
+                _ribbonComboBoxThemes.Items.Add(theme);
+            }
+
+            // set default theme
+            var customThemeIndex = _ribbonComboBoxThemes.Items.IndexOf(_customThemeName);
+            if (customThemeIndex > -1)
+            {
+                _ribbonComboBoxThemes.SelectedIndex = customThemeIndex;
+            }
+        }
+
+        #endregion
+
+        #region FlexGrid Events
+
+        private void _flex_GridChanged(object sender, GridChangedEventArgs e)
```

Comment:
method naming isn't good.

Reply:
I presume this is generated with the designer, so it is based of the control name, which is `_flex`.
Whilst it may not be mainstream to prefix UI elements with an underscore, I'm not aware of any specific naming convention.

Reply:
Yes, it was generated with designer and I also don't know may I prefix UI elements with underscore or not.
Any suggestions are welcome.